app/src/main/java/com/example/ncov19traking/ui/global/GlobalFragment.kt

```diff
@@ -106,13 +144,12 @@ class GlobalFragment : Fragment() {
 
     inner class GlobalFragmentViewHolder {
         private val view = requireView()
-        val totalCases: TextView = view.findViewById(R.id.total_cases)
-        val recovered: TextView = view.findViewById(R.id.recovered_numbers)
-        val deaths: TextView = view.findViewById(R.id.deaths_numbers)
-        val casesPercentage: TextView = view.findViewById(R.id.percentageCasesDifference)
-        val recoveredPercentage: TextView = view.findViewById(R.id.percentageRecoveredDifference)
-        val deathsPercentage: TextView = view.findViewById(R.id.percentageDeathsDifference)
+        val covidTotalCasesCard: CovidCustomCard = view.findViewById(R.id.covid_total_cases_card)
+        val covidRecoveredCard: CovidCustomCard = view.findViewById(R.id.covid_recovered_cases_card)
+        val covidDeathsCard: CovidCustomCard = view.findViewById(R.id.covid_deaths_cases_card)
         val lastUpdateLong: TextView = view.findViewById(R.id.last_update_date)
-        val progressBar: ProgressBar = view.findViewById(R.id.progressBar)
+        val progressBar: ProgressBar = view.findViewById(R.id.progress_bar)
+        val pieChart = view.findViewById<PieChart>(R.id.circular_chart)
+            .apply { PieChartConfigHelper.setupPieChart(this) }
```

Comment:
It's better but let's better not do any kind of initialization here, you are initializing on other places of the fragment as well, why not do it outside in the on view created or where you are doing the other stuff, this seems really out of place here.